### ISSUE TITLE
Hibernate Panache with UUID strategy gives weird behavior

### ISSUE THREAD

**Issue by iabughosh** [quarkusio/quarkus#14612]
I am using the below ID UUID generation strategy  : 
```
    @Id
    @Column(name = "id")
    @GeneratedValue(generator = "UUID")
    @GenericGenerator(name = "UUID",
                      strategy = "org.hibernate.id.UUIDGenerator")
    public UUID id;
```

The value being stored on DB is :

> 65f10bf1-6d25-482b-9290-cb76e5ca867f

which is fine, but when I read it from DB it is being read as 

> f10bf165-256d-2b48-9290-cb76e5ca867f

If you noticed the string it gives it back with some characters are being misplaced.

P.S. If we changed data type into 'String' everything works fine.

Expected :
Both read and write should be the same value.

**Environment (please complete the following information):**
 - Java 11
 - Quarkus 1.10.5

Regards.

**Comment by FroMage** [comment]
@Sanne @yrodiere this looks like a column mapper issue, no?

**Comment by gavinking** [comment]
@iabughosh what database is this?

**Comment by iabughosh** [comment]
Sorry forgot mentioning that,
It is MS SQL Server 2019 / version 15.

**Comment by amar-kumar-singh** [comment]
@Id
	@Column(name = "id", nullable = false, length = 64)
	@GeneratedValue(generator = "UUID")
	//@GenericGenerator(name = "system-uuid", strategy = "uuid")
	@GenericGenerator(name = "UUID",
			strategy = "org.hibernate.id.UUIDGenerator")
	public String getId() {
		return id;
	}


I am getting this exception ...

2021-01-27 19:17:27,935 ERROR [org.hib.eng.jdb.spi.SqlExceptionHelper] (executor-thread-1) ERROR: relation "t_user" does not exist
  Position: 1114
2021-01-27 19:17:27,952 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /users/2c94808976085e8501761d70b7c5000e failed, error id: 8b292a54-dd45-451a-a9b8-8b005c0f294e-1: org.jboss.resteasy.spi.UnhandledException: javax.persistence.PersistenceException: org.hibernate.exception.SQLGrammarException: could not extract ResultSet
        at org.jboss.resteasy.core.ExceptionHandler.handleApplicationException(ExceptionHandler.java:106)
        at org.jboss.resteasy.core.ExceptionHandler.handleException(ExceptionHandler.java:372)
        at org.jboss.resteasy.core.SynchronousDispatcher.writeException(SynchronousDispatcher.java:218)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:519)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$invoke$4(SynchronousDispatcher.java:261)
        at org.jboss.resteasy.core.SynchronousDispatcher.lambda$preprocess$0(SynchronousDispatcher.java:161)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.SynchronousDispatcher.preprocess(SynchronousDispatcher.java:164)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:247)
        at io.quarkus.resteasy.runtime.standalone.RequestDispatcher.service(RequestDispatcher.java:73)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.dispatch(VertxRequestHandler.java:136)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler.access$000(VertxRequestHandler.java:40)
        at io.quarkus.resteasy.runtime.standalone.VertxRequestHandler$1.run(VertxRequestHandler.java:97)
        at io.quarkus.runtime.CleanableExecutor$CleaningRunnable.run(CleanableExecutor.java:231)
        at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
Caused by: javax.persistence.PersistenceException: org.hibernate.exception.SQLGrammarException: could not extract ResultSet
        at org.hibernate.internal.ExceptionConverterImpl.convert(ExceptionConverterImpl.java:154)
        at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1602)
        at org.hibernate.query.Query.getResultList(Query.java:165)
        at io.quarkus.hibernate.orm.panache.common.runtime.CommonPanacheQueryImpl.firstResult(CommonPanacheQueryImpl.java:246)
        at io.quarkus.hibernate.orm.panache.runtime.PanacheQueryImpl.firstResult(PanacheQueryImpl.java:159)
        at com.odigo.uum.userapi.repository.UserRepository.findById(UserRepository.java:21)
        at com.odigo.uum.userapi.repository.UserRepository_ClientProxy.findById(UserRepository_ClientProxy.zig:865)
        at com.odigo.uum.userapi.service.impl.UserServiceImpl.getUserById(UserServiceImpl.java:85)
        at com.odigo.uum.userapi.service.impl.UserServiceImpl_Subclass.getUserById$$superaccessor4(UserServiceImpl_Subclass.zig:634)
        at com.odigo.uum.userapi.service.impl.UserServiceImpl_Subclass$$function$$4.apply(UserServiceImpl_Subclass$$function$$4.zig:33)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:127)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.invokeInOurTx(TransactionalInterceptorBase.java:100)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.doIntercept(TransactionalInterceptorRequired.java:32)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorBase.intercept(TransactionalInterceptorBase.java:53)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired.intercept(TransactionalInterceptorRequired.java:26)
        at io.quarkus.narayana.jta.runtime.interceptor.TransactionalInterceptorRequired_Bean.intercept(TransactionalInterceptorRequired_Bean.zig:340)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at com.odigo.uum.userapi.service.impl.UserServiceImpl_Subclass.getUserById(UserServiceImpl_Subclass.zig:591)
        at com.odigo.uum.userapi.service.impl.UserServiceImpl_ClientProxy.getUserById(UserServiceImpl_ClientProxy.zig:128)
        at com.odigo.uum.userapi.controller.UserController.getUser(UserController.java:48)
        at com.odigo.uum.userapi.controller.UserController_Subclass.getUser$$superaccessor2(UserController_Subclass.zig:384)
        at com.odigo.uum.userapi.controller.UserController_Subclass$$function$$2.apply(UserController_Subclass$$function$$2.zig:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:54)
        at io.quarkus.hibernate.validator.runtime.interceptor.AbstractMethodValidationInterceptor.validateMethodInvocation(AbstractMethodValidationInterceptor.java:69)
        at io.quarkus.hibernate.validator.runtime.jaxrs.JaxrsEndPointValidationInterceptor.validateMethodInvocation(JaxrsEndPointValidationInterceptor.java:35)
        at io.quarkus.hibernate.validator.runtime.jaxrs.JaxrsEndPointValidationInterceptor_Bean.intercept(JaxrsEndPointValidationInterceptor_Bean.zig:392)
        at io.quarkus.arc.impl.InterceptorInvocation.invoke(InterceptorInvocation.java:41)
        at io.quarkus.arc.impl.AroundInvokeInvocationContext.perform(AroundInvokeInvocationContext.java:41)
        at io.quarkus.arc.impl.InvocationContexts.performAroundInvoke(InvocationContexts.java:32)
        at com.odigo.uum.userapi.controller.UserController_Subclass.getUser(UserController_Subclass.zig:339)
        at com.odigo.uum.userapi.controller.UserController_ClientProxy.getUser(UserController_ClientProxy.zig:194)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:170)
        at org.jboss.resteasy.core.MethodInjectorImpl.invoke(MethodInjectorImpl.java:130)
        at org.jboss.resteasy.core.ResourceMethodInvoker.internalInvokeOnTarget(ResourceMethodInvoker.java:643)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTargetAfterFilter(ResourceMethodInvoker.java:507)
        at org.jboss.resteasy.core.ResourceMethodInvoker.lambda$invokeOnTarget$2(ResourceMethodInvoker.java:457)
        at org.jboss.resteasy.core.interception.jaxrs.PreMatchContainerRequestContext.filter(PreMatchContainerRequestContext.java:364)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invokeOnTarget(ResourceMethodInvoker.java:459)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:419)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:393)
        at org.jboss.resteasy.core.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:68)
        at org.jboss.resteasy.core.SynchronousDispatcher.invoke(SynchronousDispatcher.java:492)
        ... 20 more
Caused by: org.hibernate.exception.SQLGrammarException: could not extract ResultSet
        at org.hibernate.exception.internal.SQLStateConversionDelegate.convert(SQLStateConversionDelegate.java:103)
        at org.hibernate.exception.internal.StandardSQLExceptionConverter.convert(StandardSQLExceptionConverter.java:42)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:113)
        at org.hibernate.engine.jdbc.spi.SqlExceptionHelper.convert(SqlExceptionHelper.java:99)
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.extract(ResultSetReturnImpl.java:67)
        at org.hibernate.loader.Loader.getResultSet(Loader.java:2304)
        at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2057)
        at org.hibernate.loader.Loader.executeQueryStatement(Loader.java:2019)
        at org.hibernate.loader.Loader.doQuery(Loader.java:948)
        at org.hibernate.loader.Loader.doQueryAndInitializeNonLazyCollections(Loader.java:349)
        at org.hibernate.loader.Loader.doList(Loader.java:2850)
        at org.hibernate.loader.Loader.doList(Loader.java:2832)
        at org.hibernate.loader.Loader.listIgnoreQueryCache(Loader.java:2664)
        at org.hibernate.loader.Loader.list(Loader.java:2659)
        at org.hibernate.loader.hql.QueryLoader.list(QueryLoader.java:506)
        at org.hibernate.hql.internal.ast.QueryTranslatorImpl.list(QueryTranslatorImpl.java:400)
        at org.hibernate.engine.query.spi.HQLQueryPlan.performList(HQLQueryPlan.java:219)
        at org.hibernate.internal.SessionImpl.list(SessionImpl.java:1414)
        at org.hibernate.query.internal.AbstractProducedQuery.doList(AbstractProducedQuery.java:1625)
        at org.hibernate.query.internal.AbstractProducedQuery.list(AbstractProducedQuery.java:1593)
        ... 67 more
Caused by: org.postgresql.util.PSQLException: ERROR: relation "t_user" does not exist
  Position: 1114
        at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2553)
        at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2285)
        at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:323)
        at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:473)
        at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:393)
        at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:164)
        at org.postgresql.jdbc.PgPreparedStatement.executeQuery(PgPreparedStatement.java:114)
        at io.agroal.pool.wrapper.PreparedStatementWrapper.executeQuery(PreparedStatementWrapper.java:76)
        at org.hibernate.engine.jdbc.internal.ResultSetReturnImpl.extract(ResultSetReturnImpl.java:57)
        ... 82 more

**Comment by Sanne** [comment]
@amar-kumar-singh that means the table `t_user` does not exist, and seems an unrelated problem to this issue.

**Comment by amar-kumar-singh** [comment]
Hello Sanne ,

# This data source configuration is used to connect Master Db to fecth tenants and other master information
# Do not modify the keys and the structure
# multitenant: DATABASE  This will serve both schema and database level multi-tenancy
# Refer README.md for more details
quarkus:
  datasource:
    connection-timeout: 20000
    db-kind: postgresql
    driver-class-name: org.postgresql.Driver
    maximum-pool-size: 5
    idle-timeout: 30000
    minimum-idle: 5
    jdbc:
      url: jdbc:postgresql://127.0.0.1:5432/masterdb
    reactive:
      max-size: 20


    password: Odigo2020
    username: postgres
  hibernate-orm:
    database:
      generation: none
    multitenant: DATABASE
  http:
    port: 8080




  # This data source configuration is used to connect to different tenants
  # Update the values according to your databse
  # Do not modify the keys and the structure
  # "yourDbName" your db name
  # Refer README.md for more details

cron:
  time: "0  45 23 * * ?"
module:
  datasource:
    connection-timeout: 20000
    driver-class-name: org.postgresql.Driver
    idle-timeout: 30000
    jdbc:
      url: jdbc:postgresql://127.0.0.1:5432/ODIGO_CLIENT_tba01?currentSchema=odigo_user_tba01
    maximum-pool-size: 5
    minimum-idle: 3
    password: Odigo2020
    username: postgres


This one is the above configuration,  I am getting the schema information [odigo_user_tba01] from masterdb.  then we hit the schema odigo_user_tba01 for this table "t_user"

**Comment by yrodiere** [comment]
Sorry it took so long to get back to you @iabughosh, it seems this fell through the cracks.

I tried to reproduce the problem on recent versions, and could not: everything seems fine. See https://github.com/yrodiere/quarkus-playground/tree/i14612-uuid

Either this has been fixed since you reported the problem, or you made a mistake when reading the UUID from the database (which is understandable since the UUID is stored as bytes on MSSQL).

Either way, it seems we should close this issue, so I'll do it. Don't hestitate to ping me if this is still a problem, preferably with a reproducer, and we'll have another look.
